otsserver/rpc.py

```diff
@@ -214,22 +236,36 @@ def do_GET(self):
 Wallet balance: {{ balance }} BTC</br>
 </p>
 
+<hr>
+
 <p>
-You can donate to the wallet by sending funds to: {{ address }}</br>
-This address changes after every donation.
+You can donate to the wallet by sending funds to:</br>
+<img src="/qr/{{ address }}" width="250" /></br>
+<span>{{ address }}</span>
 </p>
 
+<hr>
+
+{{ #lightning_invoice }}
+<p>
+You can donate through lightning network with the following invoice:</br>
+<img src="/qr/{{ lightning_invoice }}" width="400"/></br>
```

Comment:
Maybe use those special inline-data URL's for this rather than adding another endpoint?

Reply:
If they were static for sure, but since they are dynamic the page load faster if it can load first the text and then images, they are also cacheable if they are in a separate endpoint while home page is not cacheable. (I should probably add caching info to the endpoint then!)

Reply:
The status page is cached? I don't think so/I hope not, we have informations like pending commitments which changes every second.

In the case of base64, an attacker could also load the status page continuously causing the backend to calculate the qr multiple times which is not extra cheap, unless we add a memory level cache inside the calendar which must also handle expiration of values in caches because addresses and invoices changes...

Reply:
ah ok the home page is cached, I'll change it to base64